haskell-api.md

```diff
@@ -117,6 +117,107 @@ possible. Issues include:
 If you are creating a new `PUT` consider if it could be expressed in a more
 granular `PATCH` semantic.
 
+### `PATCH` semantics
+
+Modifications to an existing resource should be made via a `PATCH` request.
+
+### Specific guidelines
+
+A well-formed `PATCH` endpoint
+
+- Should update an entity's `updatedAt` field, if present.
+- Should return an empty response or the modified resource. Ideally, if the
+  latter, the resource should be re-fetched for ease of testing and to ensure
+  the update was correctly persisted.
+- Should accept `null` to unset nullable fields, disallowing it for
+  non-nullables.
+- Should 404 if a resource with the route's identifier(s) does not exist.
+- Should not allow for update-scoping information within the `PATCH` object. If
+  scoping information is absolutely necessary, it should be include in query
+  params.
+- Should not create resources (that's the role of `POST`).
+- Should allow for fields to be optional, even allowing for entirely empty
+  `PATCH` objects.
+- Should not allow for unsupported keys.
```

Comment:
I think we should remove this point and bring it in by a separate PR with isolated discussion.

Aeson doesn't make this easy and we do it nowhere (that I know of). If we did do it, it would decrease flexibility in backwards-compatible changes. Client couldn't be updated to send new fields first, before Server is updated to handle it. This is (probably) always solvable, but it can be non-ideal, for example, if Server changes are more complex so you'd rather maintain compatibility by doing Client first and make a single change from old -> new in the Server.

Reply:
Ah, yes, right! Let me remove that.